util/pace/util/monitor/convert.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+
+from .._optional_imports import cupy
+from ..quantity import Quantity
+
+
+def convert_to_numpy(quantity: Quantity) -> np.ndarray:
+    try:
+        output = np.asarray(quantity.view[:])
+    except ValueError as err:
+        if err.args[0] == "object __array__ method not producing an array":
+            output = cupy.asnumpy(quantity.view[:])
```

Comment:
This should take `quantity.view[:]` as an argument (i.e. a cupy or numpy array), since it doesn't use any other part of Quantity and it's currently a hidden side-effect that this subsets to the compute domain. I'd suggest not type hinting it, and just calling the argument `array`.